### PR TITLE
tests: instrument user events in tests

### PR DESCRIPTION
**Pull request by Zylphrex**
This adds some spans for user events in tests.

depends on getsentry/jest-sentry-environment#3

### CODE REVIEW

File: tests/js/instrumentedEnv/userEventIntegration.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import {Hub} from '@sentry/hub';
+import {fill, isThenable, loadModule} from '@sentry/utils';
+
+const moduleName = '@testing-library/user-event';
```
Comment (evanpurkhiser) [inline tests/js/instrumentedEnv/userEventIntegration.ts:4]: why in a variable like this?

Comment (Zylphrex) [reply tests/js/instrumentedEnv/userEventIntegration.ts:4]: no particular reason, an artifact of all the ways i tried to do the monkey patching. i can inline it

File: tests/js/instrumentedEnv/userEventIntegration.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+export function instrumentUserEvent(getCurrentHub: () => Hub): void {
+  const pkg = loadModule(moduleName) as any;
```
Comment (evanpurkhiser) [inline tests/js/instrumentedEnv/userEventIntegration.ts:7]: Is the typing wrong on this? Why do you have to any it?

Comment (Zylphrex) [reply tests/js/instrumentedEnv/userEventIntegration.ts:7]: The return type here is a generic and varies depending on the package itself. I opted to just type it as `any` for simplicity.

Comment (evanpurkhiser) [reply tests/js/instrumentedEnv/userEventIntegration.ts:7]: is it actually wrong though?

Comment (Zylphrex) [reply tests/js/instrumentedEnv/userEventIntegration.ts:7]: if i remove it, i get a `Object is of type 'unknown'` on the next line at `pkg.default`

Comment (evanpurkhiser) [reply tests/js/instrumentedEnv/userEventIntegration.ts:7]: 👍 

Talked on slack.

https://github.com/getsentry/sentry-javascript/blob/f042723bb442193f03d61d4d316cc2645aad6cea/packages/utils/src/node.ts#L46-L63

This isn't typed with respect to the name of the module so it doesn't understand how to resolve the types. 🤷

**Review by billyvg — APPROVED** [review]
getsentry seems unhappy but otherwise lgtm

**Pull request merged**